src/cascadia/wt/shim.cpp

```diff
@@ -0,0 +1,28 @@
+#include <string>
+#include <filesystem>
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <wil/stl.h>
+#include <wil/resource.h>
+#include <wil/win32_helpers.h>
+
+#pragma warning(suppress : 26461) // we can't change the signature of wWinMain
+int __stdcall wWinMain(HINSTANCE, HINSTANCE, LPWSTR pCmdLine, int)
+{
+    std::filesystem::path module{ wil::GetModuleFileNameW<std::wstring>(nullptr) };
+    module.replace_filename(L"WindowsTerminal.exe");
+
+    // Synthesize a commandline
+    std::wstring cmdline{ L"wt.exe " };
+    cmdline += pCmdLine;
+
+    // Get our startup info so it can be forwarded
+    STARTUPINFOW si{};
+    si.cb = sizeof(si);
+    GetStartupInfoW(&si);
+
+    // Go!
+    wil::unique_process_information pi;
+    return CreateProcessW(module.c_str(), cmdline.data(), nullptr, nullptr, FALSE, 0, nullptr, nullptr, &si, &pi);

```

Comment:
Wait, the way this is, won't the commandline `windowsterminal.exe` processes be `windownterminal.exe wt.exe new-tab ...`?

Reply:
So actually, no! The first argument to CreateProcess is the actual executable to run, and the second is its _entire_ commandline. That includes the first argument, usually the program name, that comes in as `argv[0]`.
There are some tools (usually linux ones) that detect what identity they were launched as and do something different depending on what it was. Busybox is a great example. It’s one binary that contains sh, ls, rm, cat, ... and at runtime decides which one to be based on how you invoked it.

Reply:
Anyway, all that is possible because the first argument, usually the program name, can vary and is detectable by the application. Because of that, though, it must be specified when creating a process!